app/src/sharedTest/java/org/oppia/app/topic/train/TopicTrainFragmentTest.kt

```diff
@@ -2,36 +2,138 @@ package org.oppia.app.topic.train
+      onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+      onView(withId(R.id.topic_train_start_button)).check(matches(isClickable()))
+      onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+      onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
     }
   }
 
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_clickStartButton_skillListTransferSuccessfully() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+    onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+    onView(withRecyclerView(R.id.skill_recycler_view).atPosition(1)).perform(click())
+    onView(withId(R.id.topic_train_start_button)).check(matches(isClickable()))
+    onView(withId(R.id.topic_train_start_button)).perform(click())
+    intended(hasComponent("org.oppia.app.topic.questionplayer.QuestionPlayerActivity"))
+    intended(hasExtra("QuestionPlayerActivity.skill_id_list", skillIdListIntent))
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_configurationChange_skillsAreSelected() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+    onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    onView(
+      withRecyclerView(R.id.skill_recycler_view).atPositionOnView(
+        0,
+        R.id.skill_check_box
+      )
+    ).check(matches(isChecked()))
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_configurationChange_startButtonRemainsInactive() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_configurationChange_startButtonRemainsActive() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+    onView(withRecyclerView(R.id.skill_recycler_view).atPosition(0)).perform(click())
+    onView(withId(R.id.topic_train_start_button)).check(matches(isClickable()))
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    onView(
+      withRecyclerView(R.id.skill_recycler_view).atPositionOnView(
+        0,
+        R.id.skill_check_box
+      )
+    ).check(matches(isChecked()))
```

Comment:
Do we need to recheck it since this is verified above in its own test? We can assume that test is passing when we get here.

Reply:
It was just extra added layer of testing but definitely if we remove this than also it will work.

Final: Removed checking `checkbox` state.